lib/utils/dataStructureUtils.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+function createMapFromObject(obj) {
+    const map = new Map();
+    for (const key in obj) {
+        map.set(key, obj[key]);
+    }
+    return map;
+}
+exports.createMapFromObject = createMapFromObject;
+function isArrayOfString(x) {
+    return (Array.isArray(x) &&
+        x.every(item => Boolean(item && typeof item === "string")));
+}
+exports.isArrayOfString = isArrayOfString;
+function isArrayOfObject(x) {
+    return (Array.isArray(x) &&
+        x.every(item => Boolean(item && typeof item === "object")));
+}
+exports.isArrayOfObject = isArrayOfObject;
+function isObjectEmpty(obj) {
+    return typeof obj === "object" && !Array.isArray(obj) && obj
```

Comment:
Doesn't this project have lodash? If it has, you can use `lodash/isEmpty`

Reply:
I think it's better to minimize dependencies. What do you guys think?

Reply:
I haven't used `lodash/isEmpty` but I feel like the current one (`isObjectEmpty`) is more customized